From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: FloralDroid <opensource@floraldroid.dev>
Date: Sun, 9 Aug 2026 13:30:00 +0800
Subject: [PATCH] art: expose NativeBridge headers to Floral router

The Floral aggregate NativeBridge entry point is a platform library outside
the ART source tree. Allow it to consume the stable NativeBridge callback
header while keeping the existing ART subpackage visibility.

---
 libnativebridge/Android.bp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libnativebridge/Android.bp b/libnativebridge/Android.bp
index 8e87997..64dd0e2 100644
--- a/libnativebridge/Android.bp
+++ b/libnativebridge/Android.bp
@@ -23,6 +23,11 @@ cc_defaults {
 cc_library_headers {
     name: "libnativebridge-headers",
 
+    visibility: [
+        "//art:__subpackages__",
+        "//system/floral/nativebridge",
+    ],
+
     host_supported: true,
     export_include_dirs: ["include"],
 
-- 
2.34.1
